### PR TITLE
[rllib] Update RLlib docs and README

### PR DESCRIPTION
**Pull request by ericl**
## What do these changes do?

Updates the rllib docs.

cc @pcmoritz @richardliaw

### CODE REVIEW

**Comment by ericl** [comment]
TODO in a future PR: show some perf benchmarks and architecture diagrams.

**Review by pcmoritz — COMMENTED** [review]
We probably want to have a sphinx file for developers too so that we can have autogenerated documentation.  The README should contain a higher level description to RLlib, link to the documentation and the paper once it is up.

**Comment by ericl** [comment]
@pcmoritz that makes sense, I think we should have in the Sphinx docs (1) a user-facing RLlib page, and a (2) page with RLlib developer APIs and docs.

I'm not comfortable releasing a developer API page right now though, given that the refactoring is in-progress. How about we go with the brief technical overview in the README and do the full dev api page later?

File: doc/source/rllib.rst [hunk 1]
```diff
@@ -1,15 +1,28 @@
...
-This document describes Ray's reinforcement learning library.
-It currently supports the following algorithms:
+Ray RLlib is a reinforcement learning library that aims to provide both performance and composability.
+
+- Performance
+    - High performance algorithm implementions
+    - Pluggable distributed RL execution strategies
+
+- Composability
+    - Integration with the Ray.tune hyperparam tuning tool
```
Comment (richardliaw) [inline doc/source/rllib.rst]: Can you link to the Tune docs here?

**Review by richardliaw — COMMENTED** [review]
One other change is to add a link to Tune and RLlib on https://github.com/ray-project/ray/blob/master/README.rst so that people who stumble upon the ray repository can find RLlib easily.

**Review by richardliaw — APPROVED** [review]

**Comment by ericl** [comment]
jenkins builds are currently down
the linux travis tests all pass, but the mac tests are backlogged

**Comment by aldanor** [comment]
Minor suggestion, would it be cleaner to provide `rllib/requirements.txt` so the users can just `pip install -r rlllib/requirements.txt` as opposed to copying/pasting from the README? (Travis can use the same file too then)

**Comment by robertnishihara** [comment]
@aldanor that would be fine, though it seems cleaner to me to let users do `pip install ray[rllib]`, e.g., as in

https://github.com/openai/gym/blob/efeadfdd20992aeba69e038205af90d340c5f95b/setup.py#L9-L16

what do you think?

**Pull request merged**